batchflow/models/tf/encoder_decoder.py

```diff
@@ -4,173 +4,300 @@
-        config['body']['encoder'] = dict(base_class=ResNet18)
-        config['body']['decoder'] = dict(layout='tna', factor=8, num_stages=3)
-        config['body']['embedding'] = dict(layout='cna', filters=8)
-        config['loss'] = 'mse'
+        config['body/encoder'] = dict(base=None, num_stages=None)
+        config['body/encoder/downsample'] = dict(layout='p', pool_size=2, pool_strides=2)
+        config['body/encoder/blocks'] = dict(base=cls.default_block)
+        config['body/embedding'] = dict(base=cls.default_block)
+        config['body/decoder'] = dict(skip=True, num_stages=None, factor=None)
+        config['body/decoder/upsample'] = dict(layout='tna')
+        config['body/decoder/blocks'] = dict(base=cls.default_block)
+        config['head'] = dict(layout='c', kernel_size=1)
         return config
 
+
     def build_config(self, names=None):
         config = super().build_config(names)
         config['head']['targets'] = self.targets
         return config
 
+
     @classmethod
     def body(cls, inputs, name='body', **kwargs):
-        """ Create encoder, embedding and decoder
-
-        Parameters
-        ----------
-        inputs : tf.Tensor
-            input tensor
-        filters : tuple of int
-            number of filters in decoder blocks
-        name : str
-            scope name
-
-        Returns
-        -------
-        tf.Tensor
-        """
+        """ Create encoder, embedding and decoder. """
         kwargs = cls.fill_params('body', **kwargs)
         encoder = kwargs.pop('encoder')
-        decoder = kwargs.pop('decoder')
         embedding = kwargs.pop('embedding')
+        decoder = kwargs.pop('decoder')
 
         with tf.variable_scope(name):
+            # Encoder: transition down
             encoder_outputs = cls.encoder(inputs, **encoder, **kwargs)
 
+            # Bottleneck: working with compressed representation
             x = cls.embedding(encoder_outputs[-1], **embedding, **kwargs)
-            if x != encoder_outputs[-1]:
-                encoder_outputs += [x]
+            encoder_outputs.append(x)
 
+            # Decoder: transition up
             x = cls.decoder(encoder_outputs, **decoder, **kwargs)
         return x
 
+
     @classmethod
     def head(cls, inputs, targets, name='head', **kwargs):
-        """ Linear convolutions with kernel 1 """
+        """ Linear convolutions. """
+        kwargs = cls.fill_params('head', **kwargs)
         with tf.variable_scope(name):
             x = cls.crop(inputs, targets, kwargs['data_format'])
             channels = cls.num_channels(targets)
-            x = conv_block(x, layout='c', filters=channels, kernel_size=1, **kwargs)
+            x = conv_block(x, filters=channels, **kwargs)
         return x
 
+
     @classmethod
-    def encoder(cls, inputs, base_class, name='encoder', **kwargs):
-        """ Create encoder from a base_class model
+    def default_block(cls, inputs, name='default_block', **kwargs):
```

Comment:
```suggestion
    def block(cls, inputs, name='default_block', **kwargs):
```
It's default by default :-)